gtest/src/program.rs

```diff
@@ -0,0 +1,164 @@
+use crate::{
+    log::RunResult,
+    manager::{ExtManager, Program as InnerProgram, ProgramState},
+    system::System,
+};
+use codec::Codec;
+use gear_core::{
+    message::{Message, MessageId},
+    program::{Program as CoreProgram, ProgramId},
+};
+use path_clean::PathClean;
+use std::{cell::RefCell, env, fmt::Debug, fs, path::Path};
+
+pub trait WasmProgram: Debug {
+    fn init(&mut self, payload: Vec<u8>) -> Result<Option<Vec<u8>>, &'static str>;
+    fn handle(&mut self, payload: Vec<u8>) -> Result<Option<Vec<u8>>, &'static str>;
+    fn handle_reply(&mut self, payload: Vec<u8>) -> Result<Option<Vec<u8>>, &'static str>;
+    fn debug(&mut self, data: &str) {
+        logger::debug!(target: "gwasm", "DEBUG: {}", data);
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct ProgramIdWrapper(pub(crate) ProgramId);
+
+impl<T: Into<ProgramIdWrapper> + Clone> PartialEq<T> for ProgramIdWrapper {
+    fn eq(&self, other: &T) -> bool {
+        self.0.eq(&other.clone().into().0)
+    }
+}
+
+impl From<u64> for ProgramIdWrapper {
+    fn from(other: u64) -> Self {
+        Self(other.into())
+    }
+}
+
+impl From<[u8; 32]> for ProgramIdWrapper {
+    fn from(other: [u8; 32]) -> Self {
+        Self(other.into())
+    }
+}
+
+impl From<String> for ProgramIdWrapper {
+    fn from(other: String) -> Self {
+        other[..].into()
+    }
+}
+
+impl From<&str> for ProgramIdWrapper {
+    fn from(other: &str) -> Self {
+        let id = other.strip_prefix("0x").unwrap_or(other);
+
+        let mut bytes = [0u8; 32];
+
+        if hex::decode_to_slice(id, &mut bytes).is_err() {
+            panic!("Invalid identifier: {:?}", other)
+        }
+
+        Self(bytes.into())
+    }
+}
+
+pub struct Program<'a> {
+    pub(crate) manager: &'a RefCell<ExtManager>,
+    pub(crate) id: ProgramId,
```

Comment:
Could we provide a method to get the ID of the program (when it has been assigned automatically)?

Reply:
Done `program.id() -> ProgramId`